packages/fxa-auth-server/lib/routes/session.js

```diff
@@ -692,6 +693,92 @@ module.exports = function (
         return {};
       },
     },
+    {
+      method: 'GET',
+      path: '/session/verify/token',
+      options: {
+        ...SESSION_DOCS.SESSION_VERIFY_TOKEN_GET,
+        auth: {
+          strategy: 'sessionToken',
+        },
+        response: {
+          schema: isA.object({
+            accountStatus: isA
+              .string()
+              // account status cannot be null - if no account exists for the session token's uid
+              // for example if the account has been deleted since the session token was created
+              // the endpoint's authentication strategy will return an "INVALID_TOKEN" error
+              .valid('verified', 'unverified')
+              .required(),
+            sessionStatus: isA
+              .string()
+              .valid('verified', 'mustVerify', 'mustUpgrade', null)
+              .required(),
+            verificationMethod: isA.string().optional(),
+          }),
+        },
+      },
+      handler: async function (request) {
+        log.begin('Session.verify', request);
+
+        const sessionToken = request.auth.credentials;
+        const { uid } = sessionToken;
+
+        const account = await db.account(uid);
+
+        // Account unverified
+        if (account.primaryEmail.isVerified === false) {
+          return {
+            accountStatus: 'unverified',
```

Comment:
I think it would be nice to have an enum or just some way to reference the possible values that we can reuse in the front-end, probably added in `libs` somewhere? Relates to your "Map string values to enum types" comment/code in the FE.